tests/processor/document/test_split_into_partition.py

```diff
@@ -0,0 +1,185 @@
+    num_partitions = split_document_to_partitions_with_statistics._statistics["num_partitions"]
+    assert num_partitions == [3, 2]
+    document_lengths = split_document_to_partitions_with_statistics._statistics["document_lengths"]
+    assert document_lengths == [len(TEXT1), len(TEXT2)]
+    assert sum(document_lengths) != sum(partition_lengths)
+
+    with pytest.raises(TypeError) as e:
+        split_document_to_partitions_with_statistics.update_statistics("num_partitions", 1.0)
+        assert e.value == "type of given key str or value float is incorrect."
+
+    split_document_to_partitions_with_statistics.show_statistics()
+
+
+def test_split_document_to_partitions_with_initial_partition():
+
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        collect_statistics=True,
+        initial_partition_label="<initial_part>",
+    )
+
+    document = DocumentWithPartition(text=TEXT2)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 3
+    partition = partitions[0]
+    assert new_document.text[partition.start : partition.end] == "This is initial text."
+    assert partition.label == "<initial_part>"
+    document_lengths = split_document_to_partitions._statistics["document_lengths"]
+    partition_lengths = split_document_to_partitions._statistics["partition_lengths"]
+    assert document_lengths == [len(TEXT2)]
+    assert sum(document_lengths) == sum(partition_lengths)
+
+
+def test_split_document_to_partitions_without_label_group_id():
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_whitelist=["<start>", "<middle>", "<end>"],
+        skip_initial_partition=True,
+    )
+    document = DocumentWithPartition(text=TEXT1)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 0
+
+
+def test_get_partitions_with_matcher():
+    document = DocumentWithPartition(text=TEXT1)
+    matcher = re.compile("(<start>|<middle>|<end>)").finditer
+    partitions = []
+    for partition in get_partitions_with_matcher(
+        document=document,
+        matcher=matcher,
+        label_group_id=0,
+        label_whitelist=["<start>", "<middle>", "<end>"],
+    ):
+        assert isinstance(partition, LabeledSpan)
+        for p in partitions:
+            assert not have_overlap((p.start, p.end), (partition.start, partition.end))
+        partitions.append(partition)
+
+
+def test_split_document_to_partitions_with_no_parts_added():
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        label_whitelist=[],
+        skip_initial_partition=True,
+    )
+    document = DocumentWithPartition(text=TEXT1)
```

Comment:
maybe let's do that for all tests

Reply:
maybe I can define TEXT as variable in each test, since it is used in other places as well. If we write whole text everywhere then it'll expand the script.

Reply:
yeah, I meant that

Reply:
fixed in [d7d4e3c](https://github.com/ArneBinder/pie-utils/pull/21/commits/d7d4e3c0dbebdbd3bd104877d44c2b19cb9f53f3)

Reply:
thanks!